tools/build.py

```diff
@@ -310,52 +346,102 @@ def Main():
               project_definition.name))
 
       if is_disabled:
-        disabled_packages.append(project_definition.name)
+        disabled_projects.append(project_definition.name)
       else:
         builds.append(project_definition)
 
   if not os.path.exists(options.build_directory):
     os.mkdir(options.build_directory)
 
+  undefined_projects = set(project_names)
+  for disabled_package in disabled_projects:
+    undefined_projects.remove(disabled_package)
+
+  check_configuration = set()
+  failed_builds = set()
+  failed_downloads = set()
+  missing_build_dependencies = set()
+
   current_working_directory = os.getcwd()
   os.chdir(options.build_directory)
 
-  undefined_packages = list(project_names)
-  for disabled_package in disabled_packages:
-    undefined_packages.remove(disabled_package)
+  try:
+    for project_definition in list(builds):
+      if project_names and project_definition.name not in project_names:
+        builds.remove(project_definition)
+        continue
 
-  failed_builds = []
-  for project_definition in builds:
-    if project_names and project_definition.name not in project_names:
-      continue
+      undefined_projects.remove(project_definition.name)
 
-    if undefined_packages:
-      project_index = undefined_packages.index(project_definition.name)
-      del undefined_packages[project_index]
+      if not project_builder.Download(project_definition):
+        builds.remove(project_definition)
 
-    logging.info('Processing: {0:s}'.format(project_definition.name))
+        print('Failed downloading: {0:s}'.format(project_definition.name))
+        failed_downloads.add(project_definition.name)
```

Comment:
Why continue here? If a project can't be downloaded, don't we want to abort by default?

Reply:
Nope, build.py is supposed to continue with other builds. What happens is the next steps (build prep and build) for that project are skipped instead.